metaflow/plugins/argo/argo_client.py

```diff
@@ -0,0 +1,200 @@
+import json
+import os
+
+from metaflow.exception import MetaflowException
+
+
+class ArgoClientException(MetaflowException):
+    headline = "Argo Client error"
+
+
+class ArgoClient(object):
+    def __init__(self, namespace=None):
+
+        try:
+            from kubernetes import client, config
+        except (NameError, ImportError):
+            raise MetaflowException(
+                "Could not import module 'kubernetes'. Install kubernetes "
+                "Python package (https://pypi.org/project/kubernetes/) first."
+            )
+
+        if os.getenv("KUBERNETES_SERVICE_HOST"):
+            # We are inside a pod, authenticate via ServiceAccount assigned
+            # to us
+            config.load_incluster_config()
+        else:
+            # Use kubeconfig, likely $HOME/.kube/config
+            # TODO (savin):
+            #     1. Support generating kubeconfig on the fly using boto3
+            #     2. Support auth via OIDC -
+            #        https://docs.aws.amazon.com/eks/latest/userguide/authenticate-oidc-identity-provider.html
+            config.load_kube_config()
```

Comment:
Can we unify this logic to fetch K8S API client with what's in `kubernetes_client.py`? Mostly because I will need to make it pluggable soon, similar to how AWS client provder is pluggable.

Reply:
Yeah - I will move the logic in one single place. I haven't touched anything in the `kubernetes_*.py` world yet - some significant changes are in-flight.